source/io/module_eggShell.js

```diff
@@ -1019,7 +1020,12 @@
                 }
             };
 
-            runExecuteSlicesAsync();
+            // Queue a microtask for starting execution
+            Promise.resolve().then(function () {
```

Comment:
The microtask will get called at the end of the current JS callstack, so roughly soon after executeSlicesUntilClumpsFinished is finished.

I ran runExecuteSlicesAsync as a microtask because that function has the ability to complete and resolve synchronously. That would make two code paths possible, one where it resolves synchronously in the current JS stack or when asynchronously in a new stack.

I made the change so no one can accidently rely on synchronous execution behavior from executeSlicesUntilClumpsFinished